pkg/build/controller/common/util.go

```diff
@@ -136,3 +142,47 @@ func HasBuildPodNameAnnotation(build *buildapi.Build) bool {
 	_, hasAnnotation := build.Annotations[buildapi.BuildPodNameAnnotation]
 	return hasAnnotation
 }
+
+// ResolveValueFrom resolves valueFrom references in build environment variables
+// including references to existing environment variables, jsonpath references to
+// the build object, secrets, and configmaps.
+// The build.Strategy.BuildStrategy.Env is replaced with the resolved references.
+func ResolveValueFrom(pod *kapi.Pod, client kclientset.Interface) error {
+	var outputEnv []kapi.EnvVar
+	var allErrs []error
+
+	build, version, err := buildadmission.GetBuildFromPod(pod)
+	if err != nil {
+		return nil
```

Comment:
+1; in https://github.com/openshift/origin/blob/master/pkg/build/admission/overrides/admission.go#L41 it's `return err`